src/framework/audio/internal/platform/win/wincoreaudiodriver.cpp

```diff
@@ -51,8 +51,7 @@ struct WinCoreData {
 
 static LPWSTR stringToLPWSTR(const std::string& str)
 {
-    static std::wstring wstr = QString::fromStdString(str).toStdWString();
-    return LPWSTR(wstr.c_str());
+    return LPWSTR(QString::fromStdString(str).toStdWString().c_str());
 }
```

Comment:
I think the correct way is:
```c++
// Here
static std::wstring stringToWstring(const std::string& str)
{
    return QString::fromStdString(str).toStdWString();
}

// In CoreAudioDriver::open
...
    } else {
        std::wstring wstr = stringToWstring(outputDeviceId);
        hr = pEnumerator->GetDevice(wstr.c_str(), &pDdevice);
    }
```
The idea is that the `std::wstring` needs to be alive always while we use its `c_str()` pointer. Basically, we should never call `c_str()` on a temporary `std::string` or `std::wstring`.

Reply:
thanks, fixed